core/shared/src/main/scala/scalaz/zio/Duration.scala

```diff
@@ -0,0 +1,73 @@
+package scalaz.zio
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.{ Duration => ScalaDuration, FiniteDuration => ScalaFiniteDuration }
+import scala.math.Ordered
+
+sealed trait Duration extends Ordered[Duration] {
+
+  def +(other: Duration): Duration
+
+  def *(factor: Double): Duration
+
+  def max(other: Duration): Duration = if (this > other) this else other
+
+  def min(other: Duration): Duration = if (this < other) this else other
+
+}
+
+final object Duration {
+
+  object Finite {
```

Comment:
`finite`? (does that even matter?)

Reply:
This companion object is needed so that we can call the private constructor of `class Finite`.